sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,338 @@
+                         const sycl::context &syclContext);
+
+void free_exportable_mem(void *deviceMemory,
+                         const sycl::queue &syclQueue);
+}
+```
+
+The `free_exportable_mem` function deallocates memory, represented by the
+`void *` parameter, which has been previously allocated through
+`alloc_exportable_device_mem`.
+
+Using `free_exportable_mem` on memory allocated through any function other
+than `alloc_exportable_device_mem` results in undefined behavior.
+
+Using `free_exportable_mem` on a memory region invalidates the handle
+returned by `export_device_mem_handle` for that region. The handle must not be
+used after the memory has been freed.
+
+If any error occurs while freeing the memory, the function will throw a
+`sycl::exception` with the `errc::runtime` code.
+
+If the SYCL backend does not support exporting memory, the function will throw a
+`sycl::exception` with the `errc::feature_not_supported` code.
+
+== Issues and Limitations
+
+=== Memory Layout
+
+This extension is currently limited to exporting memory with a linear layout. It
+does not support exporting memory with a non-linear layout, such as the
+"optimal" layout which would have an equivalent in Vulkan as
+`VK_IMAGE_LAYOUT_OPTIMAL`, or in CUDA as `cudaArray`. These "optimal" layouts
+are typically optimized for texture access.
+
+The reason for this limitation is that currently, no backend supported by
+{dpcpp} supports exporting memory with a non-linear layout. This may change in
+the future, and if it does, we could then amend the extension to support
+exporting memory with a non-linear layout.
+
+=== Closing OS Handles
+
+When a call is made to `export_device_mem_handle`, the {dpcpp} implementation
+will internally create an OS specific handle to the memory region. Both CUDA and
+Level Zero allow the user to specify the type of handle to be created. However,
+this is not always respected by the Level Zero driver. For this reason, if the
+user wishes to close the OS handle returned by `export_device_mem_handle`
+without freeing the memory, they must call the appropriate OS specific API to
+close the type of handle returned by the function.
+
+When exporting a file descriptor handle on Linux, our testing has shown that the
+`close` Linux API should work.
+
+On Windows systems, the type of OS handle returned by `export_device_mem_handle`
+may not be an NT handle (e.g. it may be a KMT handle), and therefore the user
+may experience issues when trying to close the handle using the `CloseHandle`
+Windows API.
+
+The issue of closing OS handles returned by `export_device_mem_handle` is
+something we are aware of and want to address in future versions of this
+extension. Once we have a solution, we will update this specification with a
+SYCL API that will close the OS handles returned by `export_device_mem_handle`
+without freeing the memory.
+
+=== Using `sycl::malloc_device ` and `sycl::free` for exportable memory
```

Comment:
Just acknowledging that I saw this.  I'm leaving my earlier comments in place, because my feedback is still that this alternative design would be simpler for people to use, and simpler to write a specification for.